packages/react-native-web/src/modules/ImageLoader/index.js

```diff
@@ -122,9 +122,17 @@ const ImageLoader = {
     id += 1;
     const image = new window.Image();
     image.onerror = onError;
-    image.onload = (e) => {
+    image.onload = (nativeEvent) => {
       // avoid blocking the main thread
-      const onDecode = () => onLoad({ nativeEvent: e });
+      const onDecode = () => {
+        nativeEvent.source = {
+          uri: image.src,
+          width: image.naturalWidth,
+          height: image.naturalHeight
```

Comment:
Won't `image.src` always be `''`, and `image.naturalWidth` and `image.naturalHeight` will always be `0` at this point since we just created a brand new `window.Image`?

Reply:
Not sure if it's useful to add a comment about that just to be 100% clear

Reply:
> Won't `image.src` always be `''`, and `image.naturalWidth` and `image.naturalHeight` will always be `0` at this point since we just created a brand new `window.Image`?

We are inside `image.onload` - the logic here is evaluated later, when the image did load so `src` and dimensions shouldn't be missing

You can see that `Image.getSize` pretty much does the same thing 

https://github.com/Expensify/react-native-web/blob/8f4d952b7966fd6e10831d299bdf76929dac5bc5/packages/react-native-web/src/modules/ImageLoader/index.js#L93-L100

_`callback` is called `onLoad`_

Reply:
Aah yepppp that makes sense, thanks for pointing that out 👍